src/ecmult_const_impl.h

```diff
@@ -101,16 +101,19 @@ static int secp256k1_wnaf_const(int *wnaf, const secp256k1_scalar *scalar, int w
     /* 4 */
     u_last = secp256k1_scalar_shr_int(&s, w);
     do {
-        int sign;
         int even;
 
         /* 4.1 4.4 */
         u = secp256k1_scalar_shr_int(&s, w);
         /* 4.2 */
         even = ((u & 1) == 0);
-        sign = 2 * (u_last > 0) - 1;
-        u += sign * even;
-        u_last -= sign * even * (1 << w);
+        /* In contrast to the original algorithm, u_last is always > 0 and
+         * therefore we do not need to check its sign. In particular, it's easy
+         * to see that u_last is never < 0 because u is never < 0. Moreover,
+         * u_last is never = 0 because u is never even. */
+        VERIFY_CHECK(u_last > 0);
```

Comment:
```suggestion
        VERIFY_CHECK(u_last > 0);
        VERIFY_CHECK(u_last & 1 == 0);
```

Reply:
I would argue that this is excessive given that the `u_last > 0` is only there to show a single specific difference to the paper and the whole point of the algorithm is that `u` is never even. But I can see that this slightly helps review if you don't look at the paper. I added this line.

Reply:
should be `VERIFY_CHECK((u_last & 1) == 1);` ("is never even")

Reply:
Don't trust code I've written using the GitHub web interface.